c_src/rocksdb_lib.cpp

```diff
@@ -444,9 +444,31 @@ rocksdb::Status PutTerms(db_obj_resource* rdb,
 			 rocksdb::Slice* value,
 			 std::vector<std::pair<Term, std::vector<Term>>> indices
 			) {
+    // prepare any new indices
     TermPrep tp = TermPrep(indices, key);
+    // delete any indicies that may be overwritten by this write
+    TermDelete td = TermDelete(tp.cids_, key);
     rocksdb::Status status;
     rocksdb::WriteBatch batch;
+    rocksdb::ReadOptions readoptions;
+
+    //Delete any index history
```

Comment:
Can we have reusable code blocks(divide into functions)? Same code block is under DeleteTerm.

Reply:
Neither comments are objecting this commit. Just having a conversation :)